### ISSUE TITLE
Чудо-хирургия

### ISSUE THREAD

**Issue by FreimK** [WalkmanNik/Osiris-SS13-Hi-RP#4]
Все операции производились на **/mob/living/carbon/human** инструментами, появляющимися в операционных раундстартом.

## **Некорректная работа операций по пересадке/извлечению/лечению органа в торсе и голове**
В данный момент существует возможность достать сердце и лёгкие сквозь грудную клетку, а также мозг сквозь череп.

Чтобы воспроизвести баг нужно:
1) Положить человека на операционный стол
2) Целимся в грудь или голову
3) Используем **скальпель**
4) Используем **зажим** (hemostat)
5) Используем **ретрактор**
6) В открывшемся меню выбираем интересующий нас орган (сердце/лёгкие/мозг), берем в активную руку скальпель и нажимаем **separate**. Если задача - вылечить орган, то берем в руки **Advanced Trauma Kit** и жмём **Heal**
7) ???
8) При желании орган можно будет достать, взяв в активную руку **зажим** и нажав **extract**

Очевидно, что это противоречит здравому смыслу. Ожидается, что хирург сможет взаимодействовать с данными органами только после того, как вскроет грудную клетку или черепную коробку. 

Если верить [статье на вики](http://osiris13.ru/index.php/Surgery), это достигается следующими шагами:
1) Повторяем шаги 1-5
6) Используем **циркулярную пилу**
7) Используем **ретрактор**
8) Взаимодействуем с органами

Соответственно, после взаимодействий необходимо будет восстановить ребра/череп. Нужно будет добавить выполнение данных шагов:
1) Используем **ретрактор**
2) Используем **костоправ** (bone settler)
3) Снова используем **ретрактор**

## **Невозможно "спилить" RIG циркулярной пилой.**

Опять же, если верить [вики](http://osiris13.ru/index.php/Surgery), у хирурга должна быть возможность спилить RIG циркулярной пилой. Однако, если взять **циркулярную пилу**, прицелиться в торс и кликнуть по одетому в RIG и лежащему на операционном столе игроку, ваш персонаж начнет делать разрез циркулярной пилой, игнорируя RIG. 

При использовании **сварочного аппарата** всё работает как надо.

Ожидаемое поведение: при использовании **циркулярной пилы** на человеке с RIG, начинается процесс его "снятия", после которого RIG можно будет снять со спины традиционными методами. 

## **Проведение операций сквозь броню, РИГи и т.д.**

На данный момент имеется возможность проводить любые операции сквозь броню и скафандры. Ожидается, что будут проходить проверки на наличие брони или скафандра.

Тестил на нескольких РИГах и **/obj/item/clothing/suit/armor/vest/ironhammer**

Воспроизведение:
1) Кладем тяжелобронированную цель на операционный стол
2) Начинаем проводить операцию
3) ???
4) "Доктор, как вам удалось вырезать сердца четырём сербским наемникам не снимая с них брони?"

*Пока что это всё, что мне удалось найти*

**Comment by FreimK** [comment]
Дополнение.

# При вытаскивании вещей из тела пациента они пропадают.

Воспроизведение:
1) Любым удобным способом запихиваем в пациента любой предмет
2) Кладем пациента на стол, **скальпель**, **зажим**, **ретрактор**.
3) Видим объект в ui, берем в руки **зажим**
4) Жмем на **Extract**
5) ???
6) Та-дааа! Он испарился!

Предмет пропадает как из тела, так и из мира.


Уже представляю, как нюкеры, два часа потратившие на ловлю человека с вживленным в него диском и пытающиеся достать его хирургическим путем, после того, как осознали, что диск бесследно пропал взрывают корабль далеко не ядерной боеголовкой.

Скриншоты до и после того, как я вытащил диск.
![bug_1](https://user-images.githubusercontent.com/47005899/77174968-ed971780-6ad2-11ea-929e-0d90d1b545d5.PNG)
![buh_2](https://user-images.githubusercontent.com/47005899/77174970-ee2fae00-6ad2-11ea-96f6-b4b5ebb12a58.PNG)

**Comment by Terkator** [comment]
Спиливать РИГ циркуляркой такая себе затея. Только если продолжительность процедуры будет большой, тогда можно. Ибо сваркой должно быть быстрее. Но тут нужно смотреть какие характеристики нужны для спила. Вполне божет быть welding.

**Comment by FreimK** [comment]
@Terkator 
Да, на данный момент "спил" требует QUALITY_WELDING. Лично для меня гораздо привычней его спиливать именно циркулярной пилой.
